### DISCUSSION TITLE
custom components in keystone 6

### DISCUSSION THREAD

**Discussion by jerguslejko** [keystonejs/keystone#7792] [Questions]
is it possible to add a custom component to the detail page? I want to perform a custom operation on a resource defined in schema.

I think this was supported in keystone 5 but no longer in 6?

**Comment by jschuur** [comment]
I have some custom requirements for one of my lists too.

It has a field that contains the URL for a Google Sheet and I'd like to trigger importing data from that URL and then saving it to a JSON field in the same model. The catch is, that JSON  should also be editable directly as a JSON field type.

There don't seem to be any hooks that trigger before any updates to a record have been saved, so right now, my plan is to have a segmented control that you can switch to 'sync on save' and then an afterOpertion hook checks for updated fields with that value, grabs the Sheet data and saves them to the JSON field, then resets the 'sync on save' state back to something else.

It's not ideal. There is the [star rating](https://github.com/keystonejs/keystone/tree/main/examples/custom-field) sample project for custom fields, which seems to be my angle if I wanted to create a fully customised component. I haven't quite wrapped my head around it though.

**Reply by jerguslejko** [reply]
that's creative! :D I was hoping there would be an official way to do this since it seems v5 had this feature
